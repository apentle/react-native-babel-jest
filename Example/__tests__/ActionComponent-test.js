'use strict';

import React, { Text, TouchableOpacity } from 'react-native';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

jest.unmock('../ActionComponent');
const ActionComponent = require('../ActionComponent');

describe('<ActionComponent />', () => {
  it('Component response to user action', () => {
    // Full DOM rendering
    const wrapper = mount(<ActionComponent />);
    expect(wrapper.find(Text).text()).toBe('Off');

    // Snapshot testing
    expect(mountToJson(wrapper)).toMatchSnapshot();

    // Simulate user tap
    wrapper.find(TouchableOpacity).simulate('click');
    expect(wrapper.find(Text).text()).toBe('On');

    // Snapshot testing
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
