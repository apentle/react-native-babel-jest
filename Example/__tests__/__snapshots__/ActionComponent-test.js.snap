exports[`<ActionComponent /> Component response to user action 1`] = `
<ActionComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
      }
    }>
    <div
      className="View"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
        }
      }>
      <TouchableOpacity
        onPress={[Function]}>
        <div
          className="TouchableOpacity"
          onClick={[Function]}>
          <Text
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }>
            <div
              className="Text"
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                  "textAlign": "center",
                }
              }>
              Off
            </div>
          </Text>
        </div>
      </TouchableOpacity>
    </div>
  </View>
</ActionComponent>
`;

exports[`<ActionComponent /> Component response to user action 2`] = `
<ActionComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
      }
    }>
    <div
      className="View"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
        }
      }>
      <TouchableOpacity
        onPress={[Function]}>
        <div
          className="TouchableOpacity"
          onClick={[Function]}>
          <Text
            style={
              Object {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }>
            <div
              className="Text"
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                  "textAlign": "center",
                }
              }>
              On
            </div>
          </Text>
        </div>
      </TouchableOpacity>
    </div>
  </View>
</ActionComponent>
`;
